### PR TITLE
securityhub-iam-password-policy-*

### PR DESCRIPTION
**Pull request by jihygk**
Adds remediations for securityhub-iam-password-* rules

securityhub-iam-password-policy-ensure-expires
securityhub-iam-password-policy-lowercase-letter-check
securityhub-iam-password-policy-minimum-length-check
securityhub-iam-password-policy-number-check
securityhub-iam-password-policy-prevent-reuse-check
securityhub-iam-password-policy-symbol-check

### CODE REVIEW

File: auto-remediate/lambda_handler.py [hunk 1]
```diff
@@ -56,6 +56,8 @@ def remediate(self):
                             remediation = self.security_hub.s3_bucket_public_read_prohibited(config_rule_resource_id)
                         elif 's3-bucket-public-write-prohibited' in config_rule_name:
                             remediation = self.security_hub.s3_bucket_public_write_prohibited(config_rule_resource_id)
+                        elif 'iam-password-policy' in config_rule_name:
```
Comment (mlevit) [inline auto-remediate/lambda_handler.py]: @jihygk can we maybe these in alphabetical order

File: auto-remediate/security_hub_rules.py [hunk 2]
```diff
@@ -5,7 +5,40 @@
 class SecurityHubRules:
     def __init__(self, logging):
         self.logging = logging
-    
+
+    def apply_cis_password_policy(self, resource_id):
```
Comment (mlevit) [inline auto-remediate/security_hub_rules.py]: @jihygk can we rename this to `iam_password_policy` to match the Config Rule names

File: auto-remediate/security_hub_rules.py [hunk 3]
```diff
@@ -5,7 +5,40 @@
...
+                RequireLowercaseCharacters=True,
+                AllowUsersToChangePassword=True,
+                MaxPasswordAge=90,  # days
+                PasswordReusePrevention=24,  # last 24 passwords
+                HardExpiry=False
+            )
+            self.logging.info("Updated IAM password policy with CIS AWS Foundations requirements.")
+            return True
+        except:
+            self.logging.error("Could not update IAM password policy for {resource}.".format(resource=resource_id))
```
Comment (mlevit) [inline auto-remediate/security_hub_rules.py]: @jihygk Could you change this keep it consistent:

`self.logging.error("Could not update IAM password policy for '%s'." % resource_id))`

**Review by mlevit — CHANGES_REQUESTED** [review]

File: auto-remediate/security_hub_rules.py (hunk 3 above)
Comment (jihygk) [reply auto-remediate/security_hub_rules.py]: I'll refactor existing stuff to `.format`

[The `%` style is not recommended.](https://docs.python.org/3/library/stdtypes.html#printf-style-string-formatting)

File: auto-remediate/security_hub_rules.py (hunk 2 above)
Comment (jihygk) [reply auto-remediate/security_hub_rules.py]: 1afb192

File: auto-remediate/lambda_handler.py (hunk 1 above)
Comment (jihygk) [reply auto-remediate/lambda_handler.py]: 8af426d

File: auto-remediate/security_hub_rules.py (hunk 3 above)
Comment (jihygk) [reply auto-remediate/security_hub_rules.py]: Actually I refactored to [PEP 498 f-strings](https://www.python.org/dev/peps/pep-0498/) which is much better in terms of readability + performance than `str.format()` or %

**Review by mlevit — COMMENTED** [review]
Much better man. Didn't know about this method but it's super clean and easy to read.

**Review by mlevit — COMMENTED** [review]
Once you've done your testing, please ensure you update the README with the new Config Rules you've added support for.

File: auto-remediate/security_hub_rules.py [hunk 4]
```diff
@@ -5,7 +5,40 @@
...
+        1.8 - Ensure IAM password policy requires at least one number
+        1.9 - Ensure IAM password policy requires a minimum length of 14 or greater
+        1.10 - Ensure IAM password policy prevents password reuse
+        1.11 - Ensure IAM password policy expires passwords within 90 days or less
+        """
+        client = boto3.client('iam')
+
+        # TODO: better exception handling
+        try:
+            response = client.update_account_password_policy(
```
[the remark this answers is not in the document]
Comment (mlevit) [reply auto-remediate/security_hub_rules.py]: @jihygk fix this so it stops spamming us please :P

[the remark this answers is not in the document]
Comment (jihygk) [reply auto-remediate/security_hub_rules.py]: 2488491b928edf88b5de6f7eaca7f93fcfbddd4f

**Review by mlevit — APPROVED** [review]

**Pull request merged**